util/reformat-patches.sh

```diff
@@ -0,0 +1,249 @@
+"$GIT_CMD" -C "$WORKTREE_DIR" branch -f "${WORK_BRANCH_POST}" "${TAG_POST}"
+"$GIT_CMD" -C "$WORKTREE_DIR" branch -f "${WORK_BRANCH_PRE}" "${BASE_COMMIT}"
+"$GIT_CMD" -C "$WORKTREE_DIR" branch -u "${GIT_REMOTE}/${OPENSSL_BRANCH}" "${WORK_BRANCH_PRE}"
+
+# Apply the patches
+while [ 0 -lt "$#" ]; do
+    "$GIT_CMD" -C "$WORKTREE_DIR" am "$1"
+    shift
+done
+
+# Working inside the worktree from now on
+(
+cd "${WORKTREE_DIR}"
+
+# Rebase the branch 
+"$GIT_CMD" checkout "${WORK_BRANCH_PRE}"
+"$GIT_CMD" rebase "${TAG_PRE}"
+
+# Iterate over the commits and process each with clang-format
+"$GIT_CMD" log --reverse --pretty="%H" "${TAG_PRE}..${WORK_BRANCH_PRE}" \
+    | while read -r commit; do
+        "$GIT_CMD" branch -f "${PROCESS_BRANCH_PRE}" "$commit"
+        "$GIT_CMD" checkout "${PROCESS_BRANCH_PRE}"
+        printf >&2 "Processing $("$GIT_CMD" log --pretty=oneline HEAD^..HEAD)"
+        # Process only the touched files
+        "$GIT_CMD" show --pretty="" --name-status --no-renames "$commit" \
+            | while read -r line; do
+                # Skip deletions
+                [ "x${line}" = "x${line#D}" ] || continue
+
+                fname="${line#*	}"
+
+                do_process=0
+                # Process only *.c *.h *.c.in *.h.in
+                for i in ${FMT_EXTENSIONS}; do
+                    if [ "x${fname}" != "x${fname%${i}}" ]; then
+                        do_process=1
+                        break
+                    fi
+                done
+
+                # Process the exclusion list
+                for i in ${EXCLUDE_FILES}; do
+                    if [ "x${fname}" = "x${i}" ]; then
+                        do_process=0
+                        break;
+                    fi
+                done
+
+                if [ 1 = "${do_process}" ]; then
+                    printf >&2 "%s\n" "  Formatting ${fname}"
+                    "$CLANG_FMT_CMD" -i --style=file:.clang-format "$fname"
+                else
+                    printf >&2 "%s\n" "  Including ${fname} without processing"
+                fi
+                "$GIT_CMD" add "$fname"
+            done
+
+        "$GIT_CMD" commit --amend --no-edit
+        "$GIT_CMD" checkout "${WORK_BRANCH_POST}"
+
+        "$GIT_CMD" show --pretty="" --name-status --no-renames "${PROCESS_BRANCH_PRE}" \
+            | while read -r line; do
+                fname="${line#*	}"
+
+                # Process deletions
+                if [ "x${line}" 1= "x${line#D}" ]; then
+                    "$GIT_CMD" rm "$fname"
+                    continue
+                fi
+
+                "$GIT_CMD" reset "${PROCESS_BRANCH_PRE}" -- "$fname"
+            done
+
+        "$GIT_CMD" commit -C "${commit}"
+        "$GIT_CMD" reset --hard
+    done
+)
+
+# Output the patches
+mkdir -p "${OUT_DIR}"
+OUT_DIR=$(realpath "${OUT_DIR}")
+"$GIT_CMD" -C "${WORKTREE_DIR}" format-patch -o "$(pwd)/${OUT_DIR}" "${TAG_POST}..${WORK_BRANCH_POST}"
```

Comment:
I think this is correct as you have canonicalized the OUT_DIR above with realpath.

```suggestion
"$GIT_CMD" -C "${WORKTREE_DIR}" format-patch -o "${OUT_DIR}" "${TAG_POST}..${WORK_BRANCH_POST}"
```